pallets/connectors/src/lib.rs

```diff
@@ -473,6 +494,26 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Handle an incoming message
+		/// TODO(nuno): we probably need a custom origin type for these messages to ensure they have
+		/// come in through XCM. Probably even handle it in a separate pallet? For now, let's have a
```

Comment:
IMO it should eventually be part of connectors pallet as a mod

Reply:
Yeah good idea 👍🏻

Reply:
If we have more locations relying on knowing from "where" messages are coming, i.e. the outer domains, it maybe even makes sense to have something in libs. Like a pallet with no pallet but just the origin. Although, we need to add that then to the runtimes...

Reply:
@mustermeiszer I am not following this part:

> it maybe even makes sense to have something in libs. Like a pallet with no pallet but just the origin.